### ISSUE TITLE
Signatures not inserted on TB 78.3.2

### ISSUE THREAD

**Issue by stloeffler** [4ch1m/mozext#24]
Hi,

first of all: thanks a lot for this awesome extension.
Unfortunately, it is currently not working for me (presumably after an update). I am currently running TB 78.3.2 (as packaged in Ubuntu 20.10) with SignatureSwitch 2.4.0. I can manage signatures, but trying to insert it into a mail results in the following error in the javascript console:
```
Uncaught (in promise) TypeError: browser.tabs.sendMessage is not a function
    removeSignatureFromComposer moz-extension://ba4d6a3b-3cb5-4d3c-afaa-bf41abe493ca/background/background.js:436
    appendSignatureToComposer moz-extension://ba4d6a3b-3cb5-4d3c-afaa-bf41abe493ca/background/background.js:388
background.js:436:18
```
I'm not sure if this is related, but the [TB 78.4.0 release notes](https://www.thunderbird.net/en-US/thunderbird/78.4.0/releasenotes/) (dated Oct. 21, 2020) list `MailExtensions: browser.tabs.sendMessage API added`. If this is indeed the root of the problem, would it be possible to somehow implement backwards-compatibility with TB versions before 78.4.0?

**Comment by 4ch1m** [comment]
> If this is indeed the root of the problem,

Ooops... yes, you're right.
I forgot to raise the `strict_min_version` for my latest update.
`browser.tabs.sendMessage` is only available in TB 78.4.0.

> would it be possible to somehow implement backwards-compatibility with TB versions before 78.4.0?

No... that wouldn't be possible. You'll have to manually revert back to version 2.3.1 of Signature Switch. :disappointed: 
Version 2.4.0 features several major improvements using the new API; see: [changelog](http://mozext.achimonline.de/signatureswitch-me.php#changelog)

Thanks for reporting it, though.

**Comment by 4ch1m** [comment]
version 2.4.0 withdrawn from [atn](https://addons.thunderbird.net);
released 2.4.1 with fixed "min_strict_version"

**Comment by stloeffler** [comment]
Thanks. It works. Reinstalling the extension correctly gave me version 2.3.1.
